ignite/metrics/metric.py

```diff
@@ -548,9 +548,15 @@ def another_wrapper(self: Metric, *args: Any, **kwargs: Any) -> Callable:
             if len(attrs) > 0 and not self._is_reduced:
                 if ws > 1:
                     for attr in attrs:
+                        if ":" in attr:
+                            idx = attr.index(":")
+                            op = attr[idx + 1 :].strip().upper()
```

Comment:
I’m wondering if this rule should not be included in `all_reduce`...

Reply:
You mean `strip().upper()`? it did that to handle `var: SUM` or `var: SUM `  just to strip the spaces and make it uppercase to handle `var: Sum`, `var:sum` or `var: sUM` as documented [here](https://pytorch.org/tutorials/intermediate/dist_tuto.html)

Reply:
The reduce method is the `idist` wrapper. Your api handles spaces and lower/upper case variations. My suggestion was about pushing this into the wrapper. 

But on second thoughts, it is not a good idea. Impossible to know if `SUm` is a bug or what is really desired. In fact, we should provide only one way to describe a reduction : 
* `var:max`
* `var:min`
* `var:sum`
* `var:prod`

So an exception should be raised on other string.
```python
valid_ops = ["min", "max", "sum", "prod"]
if op not in valid_ops:
    raise ValueError(f"Reduction operation is not valid (expected : {valid_ops}, got: {op}")
```
What do you think ?

Reply:
Definitely we need the exception part, but i thought keeping it as flexible as we can and easier to use is a positive thing, right?

Reply:
Btw, `all_reduce()` can't handle `min, max, ...` in a lower case it must be `MIN, MAX, ..`